thunder/core/trace.py

```diff
@@ -178,9 +178,11 @@ def set_provenance(self, provenance: TraceProvenance) -> None:
     # Methods related to name construction
     #
 
-    def add_name(self, name: str) -> None:
+    def add_name(self, name: str, *, prefix: str | None = None) -> None:
+        from thunder.core.proxies import PREFIXES_ALLOW_NAME_DUPLICATES
+
         baseutils.check(
-            name not in self.names,
+            name not in self.names or (prefix is not None and prefix in PREFIXES_ALLOW_NAME_DUPLICATES),
```

Comment:
If a program calls a custom `torch.autograd.Function` that makes a tensor subclass instance from `torch.Tensor`s, the lookaside of `torch.autograd.Function` would be called. The lookaside creates a trace of that function, and the trace would have a BoundSymbol of a prim of tensor subclass ctor that calls `SubclassTensorProxy.__init__` which calls `Proxy.__init__` which calls `TraceCtx.add_name`.
The trace's bound symbols would be passed to `OpExProcessor` and their `sym.meta` would be evaluated thus that trace would try to add the same name for subclass tensor proxy at least twice. This special casing allows that duplication.